openapi/spec-generator/v1api.go

```diff
@@ -0,0 +1,279 @@
+		SchemaType:  boolSchema,
+		Description: "",
+		In:          inHeader,
+	}
+	LastContactHeader = Parameter{
+		Id:          "LastContactHeader",
+		Name:        "X-Nomad-Last-Contact",
+		SchemaType:  intSchema,
+		Description: "",
+		In:          inHeader,
+	}
+)
+
+var (
+	XNomadIndexHeader = ResponseHeader{
+		Name:        "X-Nomad-Index",
+		SchemaType:  intSchema,
+		Description: "A unique identifier representing the current state of the requested resource. On a new Nomad cluster the value of this index starts at 1.",
+	}
+	XNomadKnownLeaderHeader = ResponseHeader{
+		Name:        "X-Nomad-KnownLeader",
+		SchemaType:  boolSchema,
+		Description: "Boolean indicating if there is a known cluster leader.",
+	}
+	XNomadLastContactHeader = ResponseHeader{
+		Name:        "X-Nomad-LastContact",
+		SchemaType:  intSchema,
+		Description: "The time in milliseconds that a server was last contacted by the leader node.",
+	}
+)
+
+var queryMeta = []*ResponseHeader{
+	&XNomadIndexHeader,
+	&XNomadKnownLeaderHeader,
+	&XNomadLastContactHeader,
+}
+
+var writeMeta = []*ResponseHeader{
+	&XNomadIndexHeader,
+}
+
+var queryOptions = []*Parameter{
+	&RegionParam,
+	&NamespaceParam,
+	&IndexHeader,
+	&WaitParam,
+	&StaleParam,
+	&PrefixParam,
+	&NomadTokenHeader,
+	&PerPageParam,
+	&NextTokenParam,
+}
+
+var writeOptions = []*Parameter{
+	&RegionParam,
+	&NamespaceParam,
+	&NomadTokenHeader,
+	&IdempotencyTokenParam,
+}
+
+var (
+	BadRequestResponse = Response{
+		Name:        "BadRequestResponse",
+		Description: "Bad request",
+	}
+	ForbiddenResponse = Response{
+		Name:        "ForbiddenResponse",
+		Description: "Forbidden",
+	}
+	InternalServerErrorResponse = Response{
+		Name:        "InternalServerErrorResponse",
+		Description: "Internal server error",
+	}
+	MethodNotAllowedResponse = Response{
+		Name:        "MethodNotAllowedResponse",
+		Description: "Method not allowed",
+	}
+	NotFoundResponse = Response{
+		Name:        "NotFoundResponse",
+		Description: "Not found",
+	}
+)
+
+var standardResponses = []*ResponseConfig{
+	{
+		Code:     403,
+		Response: &ForbiddenResponse,
+	},
+	{
+		Code:     500,
+		Response: &InternalServerErrorResponse,
+	},
+	{
+		Code:     405,
+		Response: &MethodNotAllowedResponse,
+	},
+}
+
+func newOperation(method string, handler string, tags []string, operationId string, requestBody *RequestBody, params []*Parameter, responses ...*ResponseConfig) *Operation {
+	return &Operation{
+		Method:      method,
+		Handler:     handler,
+		Tags:        tags,
+		OperationId: operationId,
+		RequestBody: requestBody,
+		Parameters:  params,
+		Responses:   getResponses(responses...),
+	}
+}
+
+func newRequestBody(schemaType string, model interface{}) *RequestBody {
+	return &RequestBody{
+		SchemaType: schemaType,
+		Model:      reflect.TypeOf(model),
+	}
+}
+
+func newResponseConfig(statusCode int, schemaType string, model interface{}, headers []*ResponseHeader, name string) *ResponseConfig {
```

Comment:
> Looking at current invocation, I see `statusCode` always being 200, which seems reasonable. Does it need to be configurable?

Yes. Some endpoints return a custom 404 if say, an ID didn't find a match, and I'll need to add those responses here.

> 
> Also, `schemaType` and `model` arguments are coupled. Can we infer `schemaType` from model? For example, `schemaType` should be "", if `model` is nil, and `arraySchema` if `model` is an array? It might require changing some declarations to use slices for list queries.

Actually, SchemaType and Model are separate so that they won't be coupled. SchemaType informs shape of the ResponseRef, and the model is fed to the kin model generator, and stashed in the Components.Schemas SchemaRef map. They have discreet, separate purposes. I think I'll keep these separate keeps things simple.